### PR TITLE
Emulate app script

### PR DESCRIPTION
**Pull request by Oscar-Pepper**

### CODE REVIEW

**Comment by fluidvanadium** [comment]
i failed to run the script, i noticed i got the same error as sometimes effects integration tests:
```
./scripts/emulate_app.sh -a x86

Installing latest build tools, platform tools, and platform...
Loading package information...                                       Loading local repository...                                          [                                       ] 3% Loading local repository[                                       ] 3% Fetch remote repository.[=                                      ] 3% Fetch remote repository.[=                                      ] 4% Fetch remote repository.[=                                      ] 5% Fetch remote repository.[==                                     ] 5% Fetch remote repository.[==                                     ] 6% Fetch remote repository.[==                                     ] 7% Fetch remote repository.[==                                     ] 7% Computing updates...    [===                                    ] 8% Computing updates...    [===                                    ] 10% Computing updates...   [=======================================] 100% Computing updates...

Installing latest emulator...
Loading package information...                                       Loading local repository...                                          [                                       ] 3% Loading local repository[                                       ] 3% Fetch remote repository.[=                                      ] 3% Fetch remote repository.[=                                      ] 4% Fetch remote repository.[=                                      ] 5% Fetch remote repository.[==                                     ] 5% Fetch remote repository.[==                                     ] 6% Fetch remote repository.[==                                     ] 7% Fetch remote repository.[==                                     ] 7% Computing updates...    [===                                    ] 8% Computing updates...    [===                                    ] 10% Computing updates...   [=======================================] 100% Computing updates...

Installing system image...
Loading package information...                                       Loading local repository...                                          [                                       ] 3% Loading local repository[                                       ] 3% Fetch remote repository.[=                                      ] 3% Fetch remote repository.[=                                      ] 4% Fetch remote repository.[=                                      ] 5% Fetch remote repository.[==                                     ] 5% Fetch remote repository.[==                                     ] 6% Fetch remote repository.[==                                     ] 7% Fetch remote repository.[==                                     ] 7% Computing updates...    [===                                    ] 8% Computing updates...    [===                                    ] 10% Computing updates...   [=======================================] 100% Computing updates...

Loading local repository...                                          [=========                              ] 25% Loading local repositor[=========                              ] 25% Fetch remote repository[==========                             ] 26% Fetch remote repository[============                           ] 31% Fetch remote repository[=============                          ] 33% Fetch remote repository[=============                          ] 34% Fetch remote repository[==============                         ] 36% Fetch remote repository[==============                         ] 37% Fetch remote repository[===============                        ] 38% Fetch remote repository[===============                        ] 40% Fetch remote repository[================                       ] 41% Fetch remote repository[=================                      ] 43% Fetch remote repository[=================                      ] 44% Fetch remote repository[==================                     ] 45% Fetch remote repository[==================                     ] 47% Fetch remote repository[===================                    ] 48% Fetch remote repository[===================                    ] 50% Fetch remote repository[====================                   ] 51% Fetch remote repository[====================                   ] 53% Fetch remote repository[=====================                  ] 54% Fetch remote repository[======================                 ] 55% Fetch remote repository[======================                 ] 57% Fetch remote repository[=======================                ] 58% Fetch remote repository[=======================                ] 60% Fetch remote repository[========================               ] 61% Fetch remote repository[========================               ] 62% Fetch remote repository[=========================              ] 64% Fetch remote repository[==========================             ] 65% Fetch remote repository[==========================             ] 67% Fetch remote repository[===========================            ] 68% Fetch remote repository[===========================            ] 69% Fetch remote repository[============================           ] 71% Fetch remote repository[============================           ] 72% Fetch remote repository[=============================          ] 74% Fetch remote repository[=============================          ] 75% Fetch remote repository[=============================          ] 75% Computing updates...   [=======================================] 100% Computing updates...
All SDK package licenses accepted.


Building APKs...

> Configure project :app
WARNING: The Gradle Managed Device DSL and associated tests are experimental

> Configure project :react-native-reanimated
AAR for react-native-reanimated has been found
/home/sava/ZINGO/testing/node_modules/react-native-reanimated/android/react-native-reanimated-70-hermes.aar

> Task :app:dataBindingMergeDependencyArtifactsDebug FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:dataBindingMergeDependencyArtifactsDebug'.
> Could not resolve all files for configuration ':app:debugCompileClasspath'.
   > Could not resolve com.facebook.soloader:soloader:0.10.5+.
     Required by:
         project :app
      > Failed to list versions for com.facebook.soloader:soloader.
         > Unable to load Maven meta-data from https://repo.maven.apache.org/maven2/com/facebook/soloader/soloader/maven-metadata.xml.
            > Could not HEAD 'https://repo.maven.apache.org/maven2/com/facebook/soloader/soloader/maven-metadata.xml'.
               > repo.maven.apache.org
      > Failed to list versions for com.facebook.soloader:soloader.
         > Unable to load Maven meta-data from https://repo.maven.apache.org/maven2/com/facebook/soloader/soloader/maven-metadata.xml.
            > Could not HEAD 'https://repo.maven.apache.org/maven2/com/facebook/soloader/soloader/maven-metadata.xml'.
               > repo.maven.apache.org
   > Could not resolve com.fasterxml.jackson.module:jackson-module-kotlin:2.14.+.
     Required by:
         project :app
      > Failed to list versions for com.fasterxml.jackson.module:jackson-module-kotlin.
         > Unable to load Maven meta-data from https://repo.maven.apache.org/maven2/com/fasterxml/jackson/module/jackson-module-kotlin/maven-metadata.xml.
            > Could not HEAD 'https://repo.maven.apache.org/maven2/com/fasterxml/jackson/module/jackson-module-kotlin/maven-metadata.xml'.
               > repo.maven.apache.org
      > Failed to list versions for com.fasterxml.jackson.module:jackson-module-kotlin.
         > Unable to load Maven meta-data from https://repo.maven.apache.org/maven2/com/fasterxml/jackson/module/jackson-module-kotlin/maven-metadata.xml.
            > Could not HEAD 'https://repo.maven.apache.org/maven2/com/fasterxml/jackson/module/jackson-module-kotlin/maven-metadata.xml'.
               > repo.maven.apache.org

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.

* Get more help at https://help.gradle.org

Deprecated Gradle features were used in this build, making it incompatible with Gradle 8.0.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

See https://docs.gradle.org/7.3.3/userguide/command_line_interface.html#sec:command_line_warnings

BUILD FAILED in 16s
3 actionable tasks: 3 executed
```

File: scripts/emulate_app.sh [hunk 1]
```diff
@@ -0,0 +1,283 @@
...
+sdk="system-images;android-${api_level};${api_target};${arch}"
+sdkmanager --install "${sdk}"
+sdkmanager --licenses
+
+# Kill all emulators
+../scripts/kill_emulators.sh
+
+if [[ $create_snapshot == true ]]; then
+    echo -e "\nCreating AVD..."
+    echo no | avdmanager create avd --force --name "${avd_name}" --package "${sdk}" --device "${avd_skin}" --skin "${avd_skin}"
```
Comment (fluidvanadium) [inline scripts/emulate_app.sh:199]: the skin causes a problem for me locally. i dont have these skins downloaded

File: scripts/emulate_app.sh [hunk 2]
```diff
@@ -0,0 +1,283 @@
...
+    echo -e "\nSnapshot saved"
+
+    # Kill emulator
+    ../scripts/kill_emulators.sh
+else
+    echo -e "\nChecking for AVD..."
+    if [ $(emulator -list-avds | grep -ow "${avd_name}" | wc -w) -ne 1 ]; then
+        echo "AVD not found"
+        echo -e "\nCreating AVD..."
+        echo no | avdmanager create avd --force --name "${avd_name}" --package "${sdk}" --device "${avd_skin}" --skin "${avd_skin}"
```
Comment (fluidvanadium) [inline scripts/emulate_app.sh:221]: ^

**Review by fluidvanadium — DISMISSED** [review]
success with 4 different abis, whatever they are

File: scripts/emulate_app.sh (hunk 1 above)
Comment (Oscar-Pepper) [reply scripts/emulate_app.sh:199]: oh i dont remember manually dowloading these, it must share a folder with android studio. i will try to add something to setup skins now

**Review by fluidvanadium — APPROVED** [review]
wurx

**Pull request merged**